src/server/auth/cmds/cmds.go

```diff
@@ -99,29 +107,54 @@ Activate Pachyderm's auth system, and restrict access to existing data to the ro
 
 				c.SetAuthToken(resp.PachToken)
 			}
-			if _, err := c.PfsAPIClient.ActivateAuth(c.Ctx(), &pfs.ActivateAuthRequest{}); err != nil {
-				return errors.Wrapf(grpcutil.ScrubGRPC(err), "error configuring auth for existing PFS repos - run `pachctl auth activate` again")
+
+			// The enterprise server doesn't have PFS or PPS enabled
+			if !enterprise {
+				if _, err := c.PfsAPIClient.ActivateAuth(c.Ctx(), &pfs.ActivateAuthRequest{}); err != nil {
+					return errors.Wrapf(grpcutil.ScrubGRPC(err), "error configuring auth for existing PFS repos - run `pachctl auth activate` again")
+				}
+				if _, err := c.PpsAPIClient.ActivateAuth(c.Ctx(), &pps.ActivateAuthRequest{}); err != nil {
+					return errors.Wrapf(grpcutil.ScrubGRPC(err), "error configuring auth for existing PPS pipelines - run `pachctl auth activate` again")
+				}
+			}
+
+			// If the user wants to manually configure OIDC, stop here
+			if onlyActivate {
+				return nil
+			}
+
+			// Check whether the enterprise context is a separate identity server
+			conf, err := config.Read(false, true)
+			if err != nil {
+				return errors.Wrapf(grpcutil.ScrubGRPC(err), "failed to read configuration file")
+			}
+
+			activeContext, _, err := conf.ActiveContext(true)
+			if err != nil {
+				return errors.Wrapf(grpcutil.ScrubGRPC(err), "failed to get active context")
 			}
-			if _, err := c.PpsAPIClient.ActivateAuth(c.Ctx(), &pps.ActivateAuthRequest{}); err != nil {
-				return errors.Wrapf(grpcutil.ScrubGRPC(err), "error configuring auth for existing PPS pipelines - run `pachctl auth activate` again")
+
+			enterpriseContext, _, err := conf.ActiveEnterpriseContext(true)
+			if err != nil {
+				return errors.Wrapf(grpcutil.ScrubGRPC(err), "failed to get active context")
 			}
 
-			// By default, configure pachd as an OIDC client for the embedded Dex server.
-			// If this fails users may need to configure the server manually.
-			if !onlyActivate {
+			// In a single-node pachd deployment, configure the OIDC server.
+			// Otherwise just register with the existing identity server.
+			if activeContext == enterpriseContext || enterprise {
 				if _, err := c.SetIdentityServerConfig(c.Ctx(), &identity.SetIdentityServerConfigRequest{
 					Config: &identity.IdentityServerConfig{
-						Issuer: "http://localhost:30658/",
+						Issuer: issuer,
```

Comment:
Should we be doing any validation of this?  This is just a string argument.  Can an issuer be parsed?

Reply:
It's parsed server-side, we attempt to fetch the OIDC endpoint and refuse to update the value if it doesn't appear to be a well-behaved OIDC provider.